test/unit/org/apache/cassandra/audit/AuditLoggerCleanupTest.java

```diff
@@ -0,0 +1,132 @@
+ */
+
+package org.apache.cassandra.audit;
+
+import java.io.File;
+import java.net.InetAddress;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.config.Config;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.config.OverrideConfigurationLoader;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.service.EmbeddedCassandraService;
+import org.apache.cassandra.service.StorageService;
+
+import static java.nio.file.Files.list;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AuditLoggerCleanupTest
+{
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private EmbeddedCassandraService embedded;
+
+    private File emptyCq4File;
+
+    @Before
+    public void setup() throws Exception
+    {
+        OverrideConfigurationLoader.override((config) -> {
+            config.audit_logging_options.enabled = true;
+            config.audit_logging_options.audit_logs_dir = temporaryFolder.getRoot().getAbsolutePath();
+            config.audit_logging_options.roll_cycle = "MINUTELY";
+        });
+
+        // invalid file will be removed on startup
+        emptyCq4File = temporaryFolder.newFile("20220928-12.cq4");
+
+        CQLTester.prepareServer();
+
+        System.setProperty("cassandra.superuser_setup_delay_ms", "0");
+        embedded = new EmbeddedCassandraService();
+        embedded.start();
+    }
+
+    @After
+    public void shutdown()
+    {
+        embedded.stop();
+        System.clearProperty(Config.PROPERTY_PREFIX + "config.loader");
+    }
+
+    @Test
+    public void test() throws Throwable
+    {
+        // node started even there was empty cq4 file as it was removed upon start
+        assertTrue(StorageService.instance.isAuditLogEnabled());
+        assertFalse(emptyCq4File.exists());
+
+        insertData();
+
+        assertLogFileExists();
+
+        StorageService.instance.disableAuditLog();
+
+        // disabling and enabling from JMX will trigger same empty file cleanup
+
+        emptyCq4File = temporaryFolder.newFile("20220928-12.cq4");
+
+        StorageService.instance.enableAuditLog(null, null, null, null, null, null, null, null);
+
+        assertTrue(StorageService.instance.isAuditLogEnabled());
+
+        // invalid file were removed again
+        assertFalse(emptyCq4File.exists());
+        // only valid files are present
+        assertLogFileExists();
+    }
+
+    private void assertLogFileExists() throws Exception
+    {
+        assertTrue(list(temporaryFolder.getRoot().toPath()).allMatch(p -> {
+            String fileName = p.getFileName().toString();
+            return fileName.endsWith("cq4") || fileName.endsWith("cq4t");
+        }));
```

Comment:
nit: Getting a compiler warning around closing the `Stream`. I guess we could always do...

```
try (Stream<Path> stream = list(temporaryFolder.getRoot().toPath()))
{
    assertTrue(stream.allMatch(p -> {
        String fileName = p.getFileName().toString();
        return fileName.endsWith("cq4") || fileName.endsWith("cq4t");
    }));
}
```